tests/homework2/task2/test_cache.py

```diff
@@ -0,0 +1,218 @@
+from collections import OrderedDict
+from unittest import TestCase
+from homework.homework2.task2.cache import cache_decorator
+
+
+class CacheFunctionWithOneArgumentTestCase(TestCase):
+    def test_negative_size_cache(self):
+        with self.assertRaises(ValueError) as context:
+
+            @cache_decorator(size=-162)
+            def fib(n: int) -> int:
+                if n == 1 or n == 2:
+                    return 1
+                return fib(n - 1) + fib(n - 2)
+
+        self.assertTrue("Size must be a non-negative number." in str(context.exception))
+
+    def test_zero_size_cache(self):
+        @cache_decorator
+        def fib(n: int) -> int:
+            if n == 1 or n == 2:
+                return 1
+            return fib(n - 1) + fib(n - 2)
+
+        fib(5)
+        fib(6)
+        fib(7)
+
+        self.assertFalse(fib._cache)
```

Comment:
It looks a little bit strange since `_cache` is `OrderedDict`, but you use `assertFalse`. Maybe is it better to use `is_empty` or something like this? What do you think?